src/main/java/net/minecraftforge/common/util/RotationHelper.java

```diff
@@ -0,0 +1,98 @@
+package net.minecraftforge.common.util;
+
+import net.minecraft.block.BlockQuartz;
+import net.minecraft.block.BlockLog.EnumAxis;
+import net.minecraft.block.properties.IProperty;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.util.BlockPos;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.world.World;
+
+public class RotationHelper
+{
+
+    static public EnumAxis fromQuartz( BlockQuartz.EnumType type )
+    {
+        switch(type)
+        {
+            case LINES_X:
+                return EnumAxis.X;
+            case LINES_Y:
+                return EnumAxis.Y;
+            case LINES_Z:
+                return EnumAxis.Z;
+            default:
+                return EnumAxis.NONE;
+        }
+    }
+    
+    @SuppressWarnings( "unchecked" )
+    static public boolean rotateVanillaBlock(World world, BlockPos pos, EnumFacing axis)
+    {
+        IBlockState state = world.getBlockState(pos);
+        
```

Comment:
Wouldn't this break the intended behavior of any mods that attempt to inherit from any of these block types?

The list of block types for the first check is also fairly significant, and previously had no specific requirements.